genesis/process/argGenesisBlockCreator.go

```diff
@@ -0,0 +1,32 @@
+package process
+
+import (
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/data/state"
+	"github.com/ElrondNetwork/elrond-go/data/typeConverters"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever"
+	"github.com/ElrondNetwork/elrond-go/hashing"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/process/economics"
+	"github.com/ElrondNetwork/elrond-go/sharding"
+)
+
+// ArgsGenesisBlockCreator holds the arguments which are needed to create a genesis metablock
+type ArgsGenesisBlockCreator struct {
+	GenesisTime              uint64
+	StartEpochNum            uint32
+	Accounts                 state.AccountsAdapter
+	PubkeyConv               state.PubkeyConverter
+	InitialNodesSetup        InitialNodesHandler
+	Economics                *economics.EconomicsData
```

Comment:
maybe change it to an interface

Reply:
I tried to do that bu it seems that the pointer is passed to other factories. Will add a TODO to refactor in another PR